tools/run-integration-tests.py

```diff
@@ -0,0 +1,182 @@
+                        for envvar in credentials[cloud]:
+                            env[envvar] = credentials[cloud].get(envvar)
+                # Tox command itself
+                command = "tox -e behave-{}-{}".format(
+                    p, series_version
+                ).split()
+
+                # Wip
+                if wip:
+                    command.extend(["--tags='wip'", "--stop"])
+
+                commands.append((command, env))
+
+    return commands
+
+
+@click.command()
+@click.option(
+    "-p",
+    "--platform",
+    type=click.Choice(PLATFORMS),
+    default=PLATFORMS,
+    multiple=True,
+    help="Platform to run the tests on (requires credentials file for the clouds)",
+)
+@click.option(
+    "-s",
+    "--series",
+    type=click.Choice(SERIES_TO_VERSION.keys()),
+    default=SERIES_TO_VERSION.keys(),
+    multiple=True,
+    help="Series to run the tests on",
+)
+@click.option(
+    "--token",
+    "-t",
+    type=click.Choice(TOKEN_TO_ENVVAR.keys()),
+    default=TOKEN_TO_ENVVAR.keys(),
+    multiple=True,
+    help="Tokens to use for the tests (require tokens in the credentials file)",
+)
+@click.option(
+    "--proposed",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Run tests based on existing packages in the Proposed pocket",
+)
+@click.option(
+    "--check-version",
+    type=str,
+    help="Check for a specific version in the tests",
+)
+@click.option(
+    "--credentials-path",
+    "-c",
+    type=click.Path(dir_okay=False),
+    default="tools/ua-test-credentials.yaml",
+    help="Path to the file containing the credentials to run the test",
+)
+@click.option(
+    "-w",
+    "--wip",
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Run only tests with the @wip decorator",
+)
+def run_tests(
+    platform, series, proposed, check_version, token, credentials_path, wip
+):
+    commands = build_commands(
+        platform, series, proposed, check_version, token, credentials_path, wip
+    )
+    current_datetime = datetime.now().strftime(format="%b-%d-%H%M")
+    output_dir = "test-results/{}".format(current_datetime)
+    os.makedirs(output_dir, exist_ok=True)
+    processes = []
+    for command, env in commands:
+        print("Running {}".format(command))
+        with open(
+            "{}/{}.txt".format(output_dir, command[2]), "wb"
+        ) as result_file:
+            process = subprocess.Popen(
+                command, env=env, stdout=result_file, stderr=stdout
+            )
+            processes.append(process)
+
+    while(processes):
+        for process in processes:
+            result = process.poll()
+            if result is not None:
+                if result == 0:
+                    print("{} finished sucessfully".format(process.args))
+                else:
+                    print("Failing tests for {}".format(process.args))
+                processes.remove(process)
+        time.sleep(5)
+
+    if proposed:
```

Comment:
Should we do that only if all tests succeeded ?

Reply:
I thought about this.
I had it this way in the start, but then I dropped it. Now that we don't delete test results automatically, I think yes, it makes sense.